model/annotations/task_annotations.go

```diff
@@ -52,3 +63,77 @@ func GetLatestExecutions(annotations []TaskAnnotation) []TaskAnnotation {
 	}
 	return res
 }
+
+func MoveIssueToSuspectedIssue(taskId string, execution int, issue IssueLink, username string) error {
+	newIssue := issue
+	newIssue.Source = &Source{Requester: UIRequester, Author: username, Time: time.Now()}
+	return db.Update(
+		Collection,
+		ByTaskIdAndExecution(taskId, execution),
+		bson.M{
+			"$pull": bson.M{IssuesKey: issue},
+			"$push": bson.M{SuspectedIssuesKey: newIssue}},
+	)
+}
+
+func MoveSuspectedIssueToIssue(taskId string, execution int, issue IssueLink, username string) error {
+	newIssue := issue
+	newIssue.Source = &Source{Requester: UIRequester, Author: username, Time: time.Now()}
+	return db.Update(
+		Collection,
+		ByTaskIdAndExecution(taskId, execution),
+		bson.M{
+			"$pull": bson.M{SuspectedIssuesKey: issue},
+			"$push": bson.M{IssuesKey: newIssue}},
+	)
+}
+
+func AddIssueToAnnotation(taskId string, execution int, issue IssueLink, username string) error {
+	issue.Source = &Source{
+		Author:    username,
+		Time:      time.Now(),
+		Requester: UIRequester,
+	}
+
+	_, err := db.Upsert(
+		Collection,
+		ByTaskIdAndExecution(taskId, execution),
+		bson.M{
+			"$push": bson.M{IssuesKey: issue},
+		},
+	)
+	return errors.Wrapf(err, "problem adding task annotation issue for task '%s'", taskId)
+}
+
+func AddSuspectedIssueToAnnotation(taskId string, execution int, issue IssueLink, username string) error {
+	issue.Source = &Source{
+		Author:    username,
+		Time:      time.Now(),
+		Requester: UIRequester,
+	}
+
+	_, err := db.Upsert(
+		Collection,
+		ByTaskIdAndExecution(taskId, execution),
+		bson.M{
+			"$push": bson.M{SuspectedIssuesKey: issue},
+		},
+	)
+	return errors.Wrapf(err, "problem adding task annotation suspected issue for task '%s'", taskId)
```

Comment:
Will this return an error even when the error is nil?

Reply:
Nope!